diesel/src/pg/metadata_lookup.rs

```diff
@@ -53,10 +53,17 @@ fn lookup_type(
     cache_key: &PgMetadataCacheKey<'_>,
     conn: &PgConnection,
 ) -> QueryResult<InnerPgTypeMetadata> {
+    const TEMP_SCHEMA_NAME: &str = "pg_temp";
     let search_path: String;
-
-    let search_schema = if let Some(schema) = cache_key.schema.as_ref() {
-        vec![schema.as_ref()]
+    let mut search_path_has_temp_schema = false;
+
+    let search_schema = if let Some(schema) = cache_key.schema.as_deref() {
```

Comment:
I would prefer to use a match statement for this instead, so that we don't have that nested if here.

Reply:
hmm as far as I'm concerned I like the clear distinction between "infer schema" and "provided schema".
I see "provided schema" possibly being `pg_temp` as a sub-part of that.

That's fine for me if you still want me to change though.

Reply:
I think that distinction remains clear with the following pattern:

```rust
match cache_key.schema.as_deref() {
   Some(pg_temp) => {...}
   Some(schema) => {...}
   None => { infer schema stuff here}
}
```